platform/engineering/request-for-comment/2020-11-4-storybook-hosting.md

```diff
@@ -0,0 +1,70 @@
+# RFC: Host Storybook on design.va.gov
+
+- Date: 2020-11-4
+- Related Issue: https://github.com/department-of-veterans-affairs/va.gov-team/issues/15068
+
+## Background
+We have component library documentation in two places:
+- https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/visual-design
+- https://design.va.gov/components
+
+We're going to use Storybook to consolidate the documentation into one place:
+design.va.gov.
+
+design.va.gov is a Jekyll site. The documentation in it is manually updated. The
+repo can be found at
+[`vets-design-system-documentation`](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/).
+It's deployed daily by <this Jenkins job> to the S3 bucket from which the static
+site is served.
+
+## Motivation
+We want to serve the component library documentation from Storybook on
+design.va.gov to ensure the documentation is in a single place that's
+automatically updated when updates are made to the library. We chose Storybook
+as the tool to keep the documentation closely tied to the code itself.
+
+## Scope
+Outside the scope of this document is the code location of the component library
+and Storybook. While the location of the code will impact where the CI
+configuration is kept, the solution outlined here must be possible whether the
+code lives in
+[`veteran-facing-services-tools`](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools),
+[`vets-website`](https://github.com/department-of-veterans-affairs/vets-website),
+or its own repo, as we may move it later.
+
+For now, however, we're planning on moving the component library to its own repo.
```

Comment:
Oops, I read things out of order and asked a question about this below. Please treat replying to it as ultra low priority.

Reply:
I did some noodling on this over the weekend, and while I still plan to move this to its own repo, I think we're gonna push that off for now so we can finish with the Storybook work this quarter. I hate to do it, because working in this repo is super annoying, but this is a quality-of-life improvement that won't really affect our usage of Storybook.